value/src/main/java/com/google/auto/value/AutoValueExtension.java

```diff
@@ -0,0 +1,100 @@
+package com.google.auto.value;
+
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * An AutoValueExtension allows for extra functionality to be created during the generation
+ * of an AutoValue class.
+ *
+ * Extensions are discovered at compile time using the {@link java.util.ServiceLoader} APIs,
```

Comment:
Can you insert a `<p>` at the start of the first line of each paragraph here (except the first)? Like this:
```java
 * <p>Extensions are discovered...
```